smart-contracts/src/Challenge_6.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity >=0.6.0 <0.9.0;
+
+import '@openzeppelin/contracts/token/ERC721/ERC721.sol';
+
+/// @title Challenge 6
+/// @author Byont Labs
+/// @notice A barebones NFT contract example with documentation for learning purposes
+/**
+  TODO: Implement a function called setBaseUri() that sets _baseURIextended. 
+        The function should take a string.
+  TODO: Implement a function called tokenURI() that constructs, then returns a token URI for a given token ID
+        The function should take a uint for the token id 
+        The function should override the default ERC721 tokenURI() function
+        The function should construct the token URI by concatenating the baseURI with the given token ID
+        The function should return the token ID
+ */
+contract Contract is ERC721 {
+  using Strings for uint256;
+
+  constructor(
+    string memory _name,
+    string memory _symbol,
+    uint256 _maxSupply
+  ) ERC721(_name, _symbol) {
+    maxSupply = _maxSupply;
+  }
+
+  /// @notice The max amount of allowed NFT mints
+  uint256 public maxSupply;
+
+  /// @notice The base uri for NFT metadata, should become something like ipfs://somestring/
+  /// @dev Private strings are NOT hidden for humans, just for other smart contracts
+  string private _baseURIextended;
+
+  /// TODO: Write setBaseURI here
```

Comment:
Have you already introduced the concept of access guards at this stage? Seems good to me to include it (I think early on), and from then on automatically have it come back for all relevant parts. So they do it right from the get go. 

E.G. Have them implement an ownable in the beginning, and in later parts just include ownable library from openzeppelin, and when providing a challenge add should only be able to be called by the owner